src/aten/sycl/CumsumKernel.cpp

```diff
@@ -0,0 +1,38 @@
+// #define TORCH_ASSERT_NO_OPERATORS
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+
+#include <aten/sycl/ScanKernels.h>
+#include <aten/sycl/ScanUtils.h>
+
+namespace at::native {
+
+template <typename In, typename Out>
+struct ScalarConvert {
+  static Out to(const In v) {
+    return (Out)v;
+  }
+};
+
+namespace xpu {
+
+void launch_cumsum_xpu_kernel(
+    const Tensor& out,
+    const Tensor& self,
+    int64_t dim) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND2(
+      ScalarType::Half,
+      ScalarType::BFloat16,
+      out.scalar_type(),
+      "cumsum_xpu",
+      [&]() {
+        scan<INCLUSIVE_TYPE, scalar_t, scalar_t>(
+            out,
+            self,
+            dim,
+            ScalarConvert<float, scalar_t>::to(0.0),
```

Comment:
Just an explicit conversion of basic data type? Remove the helper,
`scalar_t(0.0f)`

Reply:
Or following CUDA,
<img width="761" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/58fcccaf-ec88-4121-b8b1-b9b9d83a9d35">